src/Workspaces/Core/Portable/Editing/SyntaxGenerator.cs

```diff
@@ -223,38 +244,10 @@ public SyntaxNode OperatorDeclaration(IMethodSymbol method, IEnumerable<SyntaxNo
             return decl;
         }
 
-        private static OperatorKind GetOperatorKind(IMethodSymbol method)
-            => method.Name switch
-            {
-                WellKnownMemberNames.ImplicitConversionName => OperatorKind.ImplicitConversion,
-                WellKnownMemberNames.ExplicitConversionName => OperatorKind.ExplicitConversion,
-                WellKnownMemberNames.AdditionOperatorName => OperatorKind.Addition,
-                WellKnownMemberNames.BitwiseAndOperatorName => OperatorKind.BitwiseAnd,
-                WellKnownMemberNames.BitwiseOrOperatorName => OperatorKind.BitwiseOr,
-                WellKnownMemberNames.DecrementOperatorName => OperatorKind.Decrement,
-                WellKnownMemberNames.DivisionOperatorName => OperatorKind.Division,
-                WellKnownMemberNames.EqualityOperatorName => OperatorKind.Equality,
-                WellKnownMemberNames.ExclusiveOrOperatorName => OperatorKind.ExclusiveOr,
-                WellKnownMemberNames.FalseOperatorName => OperatorKind.False,
-                WellKnownMemberNames.GreaterThanOperatorName => OperatorKind.GreaterThan,
-                WellKnownMemberNames.GreaterThanOrEqualOperatorName => OperatorKind.GreaterThanOrEqual,
-                WellKnownMemberNames.IncrementOperatorName => OperatorKind.Increment,
-                WellKnownMemberNames.InequalityOperatorName => OperatorKind.Inequality,
-                WellKnownMemberNames.LeftShiftOperatorName => OperatorKind.LeftShift,
-                WellKnownMemberNames.LessThanOperatorName => OperatorKind.LessThan,
-                WellKnownMemberNames.LessThanOrEqualOperatorName => OperatorKind.LessThanOrEqual,
-                WellKnownMemberNames.LogicalNotOperatorName => OperatorKind.LogicalNot,
-                WellKnownMemberNames.ModulusOperatorName => OperatorKind.Modulus,
-                WellKnownMemberNames.MultiplyOperatorName => OperatorKind.Multiply,
-                WellKnownMemberNames.OnesComplementOperatorName => OperatorKind.OnesComplement,
-                WellKnownMemberNames.RightShiftOperatorName => OperatorKind.RightShift,
-                WellKnownMemberNames.UnsignedRightShiftOperatorName => OperatorKind.UnsignedRightShift,
-                WellKnownMemberNames.SubtractionOperatorName => OperatorKind.Subtraction,
-                WellKnownMemberNames.TrueOperatorName => OperatorKind.True,
-                WellKnownMemberNames.UnaryNegationOperatorName => OperatorKind.UnaryNegation,
-                WellKnownMemberNames.UnaryPlusOperatorName => OperatorKind.UnaryPlus,
-                _ => throw new ArgumentException("Unknown operator kind."),
-            };
+        private protected virtual int GetOperatorSyntaxKind(IMethodSymbol method, out bool isChecked)
```

Comment:
why virtual and not abstract (applies to all methods like this).

Reply:
@CyrusNajmabadi SyntaxGenerator is public. Adding new abstract members is breaking.

Reply:
SyntaxGenerator has internal-abstracts.  It cannot be publicly subclassed :)  you're fine adding a new abstract to it.